arch/arm64: cache: dc_val should be uint64_t, not uint32_t in arm64_dcache_all()

### Describe the bug

In `include/zephyr/arch/arm64/cache.h`, the variable `dc_val` in `arm64_dcache_all()` is declared as `uint32_t` but is passed to the `DC` instruction via inline assembly with the `"r"` constraint.

On AArch64, the `"r"` constraint selects:
- A **32-bit W register** when the C type is `uint32_t`
- A **64-bit X register** when the C type is `uint64_t`

The ARM architecture requires the `DC ISW`, `DC CISW`, and `DC CSW` instructions to take a **64-bit X register** (`Xt`) as their operand. Using `uint32_t` causes the compiler to emit `dc isw, w<N>` (wrong) instead of `dc isw, x<N>` (correct).

Clang detects this with `-Wasm-operand-widths` and fails the build. GCC silently accepts it (no equivalent warning), which is why this was not caught in upstream CI.

### Root Cause

The `DC <dc_op>, Xt` instruction requires a **64-bit general-purpose register** (`Xt`) as its operand, as specified in the ARM Architecture Reference Manual (DDI0487).
https://support.arm.com/documentation/ddi0487/mc/-Part-C-The-AArch64-Instruction-Set/-Chapter-C5-The-A64-System-Instruction-Class/-C5-3-A64-System-instructions-for-cache-maintenance

### Regression

- [ ] This is a regression.

### Steps to reproduce

1. Check out Zephyr `main` 
2. Enable `CONFIG_DCACHE=y` and `CONFIG_ARM64_DCACHE_ALL_OPS=y` in your build configuration.
3. Build any ARM64 board target using a **Clang** toolchain with `-Wasm-operand-widths` enabled.
4. Observe three build errors at `include/zephyr/arch/arm64/cache.h` lines 199, 201, and 203 - one for each `dc_ops()` call inside `arm64_dcache_all()`.
5. Repeat the same build using **GCC** -i t succeeds with zero warnings, confirming GCC does not detect the type mismatch.

### Relevant log output

```shell
Reproduced with **Clang 19.1.7**, `--target=aarch64-none-elf`, flags `-Wasm-operand-widths -Werror`:


include/zephyr/arch/arm64/cache.h:199:20: error: value size does not match register size specified by the constraint and modifier [-Werror,-Wasm-operand-widths]
  199 |                                         dc_ops("isw", dc_val);
      |                                                       ^
include/zephyr/arch/arm64/cache.h:199:6: note: use constraint modifier "w"
include/zephyr/arch/arm64/cache.h:48:32: note: expanded from macro 'dc_ops'
   48 |         __asm__ volatile ("dc " op ", %0" :: "r" (val) : "memory");
      |                                       ^
include/zephyr/arch/arm64/cache.h:201:21: error: value size does not match register size specified by the constraint and modifier [-Werror,-Wasm-operand-widths]
  201 |                                         dc_ops("cisw", dc_val);
      |                                                        ^
include/zephyr/arch/arm64/cache.h:203:20: error: value size does not match register size specified by the constraint and modifier [-Werror,-Wasm-operand-widths]
  203 |                                         dc_ops("csw", dc_val);
      |                                                       ^
3 errors generated
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

| Item | Version |
|------|---------|
| **OS** | Ubuntu 22.04.5 LTS |
| **Zephyr version** | Current main |
| **Zephyr SDK** | 1.0.0 |
| **Clang (reproduces bug)** | 19.1.7 
| **GCC (silent pass)** | aarch64-zephyr-elf-gcc (Zephyr SDK 1.0.0) 14.3.0 |

### Additional Context

_No response_

--- a/include/zephyr/arch/arm64/cache.h
+++ b/include/zephyr/arch/arm64/cache.h
@@ -151,7 +151,8 @@ static ALWAYS_INLINE int arm64_dcache_all(int op)
 {
 	uint32_t clidr_el1, csselr_el1, ccsidr_el1;
 	uint8_t loc, ctype, cache_level, line_size, way_pos;
-	uint32_t max_ways, max_sets, dc_val, set, way;
+	uint32_t max_ways, max_sets, set, way;
+	uint64_t dc_val;
 
 	if (op != K_CACHE_INVD && op != K_CACHE_WB && op != K_CACHE_WB_INVD) {
 		return -ENOTSUP;

